tests/pass-dep/libc/libc-time.rs

```diff
@@ -95,20 +107,172 @@ fn test_localtime_r() {
         target_os = "freebsd",
         target_os = "android"
     ))]
-    assert_eq!(tm.tm_gmtoff, 0);
+    {
+        assert_eq!(tm.tm_gmtoff, 0);
+        unsafe {
+            assert_eq!(std::ffi::CStr::from_ptr(tm.tm_zone).to_str().unwrap(), "+00");
+        }
+    }
+
+    // The returned value is the pointer passed in.
+    assert!(ptr::eq(res, &mut tm));
+
+    // Remove timezone setting.
+    env::remove_var(key);
+}
+
+// PST timezone test (testing different timezone handling).
+fn test_localtime_r_pst() {
+    let key = "TZ";
+    env::set_var(key, "PST8PDT");
+    const TIME_SINCE_EPOCH: libc::time_t = 1712475836;
+    let custom_time_ptr = &TIME_SINCE_EPOCH;
+    let mut tm = create_empty_tm();
+
+    let res = unsafe { libc::localtime_r(custom_time_ptr, &mut tm) };
+
+    assert_eq!(tm.tm_sec, 56);
+    assert_eq!(tm.tm_min, 43);
+    assert_eq!(tm.tm_hour, 0); // 7 - 7 = 0 (PDT offset)
+    assert_eq!(tm.tm_mday, 7);
+    assert_eq!(tm.tm_mon, 3);
+    assert_eq!(tm.tm_year, 124);
+    assert_eq!(tm.tm_wday, 0);
+    assert_eq!(tm.tm_yday, 97);
+    assert_eq!(tm.tm_isdst, -1); // DST information unavailable
```

Comment:
If you want, `tm` has `PartialEq`, so you could compare all the fields with one assertion. Sometimes that produces hard-to-read output when the test fails though.

Reply:
Hi @saethlin,

Thank you for your comment!

The decision to use field-wise assertions was mainly driven by two factors:

* Minimizing deviation from the existing implementation.
* As you pointed out, field-wise assertions make it easier to identify the specific cause of a test failure, whereas a single assertion comparing all fields can sometimes result in hard-to-read error messages.

It's a tradeoff, but I’m open to revisiting it. Would you prefer switching to a single assertion to compare all fields using PartialEq, or do you think the current approach works better for readability and debugging?

Looking forward to your thoughts!

Reply:
Whichever you prefer is fine with me. Just making sure you'd thought about this, and clearly you have :)